packages/next/shared/lib/loadable.js

```diff
@@ -136,12 +136,19 @@ function createLoadableComponent(loadFn, options) {
       subscription.getCurrentValue
     )
 
+    // Ref to pass to the inner components
+    const innerRef = React.useRef()
+
+    // Modify the instance exposed to parents prior to being loaded
     React.useImperativeHandle(
       ref,
-      () => ({
-        retry: subscription.retry,
-      }),
-      []
+      () =>
+        state.loaded
+          ? innerRef.current
+          : {
+              retry: subscription.retry,
+            },
+      [state.loaded]
     )
```

Comment:
Theoretically, this is a breaking change: `ref.retry` is only available before the dynamic imported component is loaded, when previously `ref.retry` is always available even after the component is loaded.

It is very unlikely that someone is relying on the behavior, but we should be careful on this.